poetry/core/masonry/builders/builder.py

```diff
@@ -268,3 +302,38 @@ def temporary_directory(cls, *args, **kwargs):
             yield name
 
             shutil.rmtree(name)
+
+
+class BuildIncludeFile:
+    def __init__(
+        self,
+        path,  # type: Path
+        source_root=None,  # type: Optional[Path]
+    ):
+        """
+        :param path: a full path to the file to be included
+        :param source_root: the root path to resolve to
+        """
+        self.path = Path(path).resolve()
```

Comment:
Should resolve be done after we check for source root? Since if source root was `/tmp/build` and path is `./somefile.txt`, this would resolve to `/home/user/source/somefile.txt` if current directory is `/home/user/source`.